### PR TITLE
TIM-628 add codeartifact token

### PR DESCRIPTION
**Pull request by elianalopezv**
- Add codeartifact token to be able to get BCM packages from build actions

### CODE REVIEW

File: .github/workflows/reusable-docker-build.yaml [hunk 1]
```diff
@@ -58,9 +58,10 @@ jobs:
           role-skip-session-tagging: true
           role-to-assume: arn:aws:iam::${{ secrets.aws_account }}:role/${{ secrets.aws_assume_role }}
 
-      # - name: Test environment is configured with AWS credentials
-      #   run: |
-      #     aws sts get-caller-identity
+      - name: Set up codeartifact
+        run: |
+          CODEARTIFACT_AUTH_TOKEN=$(aws codeartifact get-authorization-token --domain bb-maven-repo --domain-owner 972594475906 --query authorizationToken --output text)
+          echo "CODEARTIFACT_AUTH_TOKEN=$CODEARTIFACT_AUTH_TOKEN" >> $GITHUB_ENV
```
Comment (toly-k) [inline .github/workflows/reusable-docker-build.yaml:64]: Is this going to dump it to the log stream?

Comment (elianalopezv) [reply .github/workflows/reusable-docker-build.yaml:64]: I think it only writes it to Github env, for example here https://github.com/MonitexInc/bmi-domain/actions/runs/5255381536/jobs/9495220460

File: .github/workflows/reusable-docker-build.yaml [hunk 2]
```diff
@@ -104,7 +105,9 @@ jobs:
       - name: Docker Build and Push local
         id: docker_build
         run: |
-          docker build . -t localhost:5000/${ECR_REPOSITORY}:${{ inputs.deployment_tag }} --build-arg MAVEN_GITHUB_TOKEN=${{ secrets.maven_github_token }}
+          docker build . -t localhost:5000/${ECR_REPOSITORY}:${{ inputs.deployment_tag }} \
+            --build-arg MAVEN_GITHUB_TOKEN=${{ secrets.maven_github_token }} \
+            --build-arg CODEARTIFACT_AUTH_TOKEN=${{ env.CODEARTIFACT_AUTH_TOKEN }}
```
[the remark this answers is not in the document]
Comment (elianalopezv) [reply .github/workflows/reusable-docker-build.yaml:110]: No, I think maven_github_token is to get them from github packages, not from codeartifact

[the remark this answers is not in the document]
Comment (elianalopezv) [reply .github/workflows/reusable-docker-build.yaml:110]: This is what I'm trying to run https://github.com/MonitexInc/bitbuy-api-admin/actions/runs/5514072797

**Review by toly-k — APPROVED** [review]

**Pull request merged**